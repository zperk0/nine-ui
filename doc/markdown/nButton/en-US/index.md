# Button
To trigger an operation.

## When To Use
A button means an operation (or a series of operations). Clicking a button will trigger corresponding business logic.

## Examples

### Button type
:::demo There are five types of buttons: default button, home button, link button, ghost button, delicate button.
```html
<template>
   <n-button>Default</n-button>
    <n-button type="primary" @click="handleClick">
      Primary
    </n-button>
    <n-button type="link">
      Link
    </n-button>
    <n-button type="ghost">
      Ghost
    </n-button>
    <n-button type="delicate">
      Delicate
    </n-button>
</template>
<script>
export default {
  methods: {
    handleClick () {
      alert('Ninecat')
    }
  }
}
</script>

```
:::

### Icon button

:::demo When you need to embed an Icon in a Button, you can set the icon property or use the Icon component directly in the Button.
```html
<template>
   <n-button icon="icon-home" />
   <n-button><n-icon name="icon-home" /></n-button>
   <n-button icon="icon-home" type="primary"/>
   <n-button icon="icon-home" type="link"/>
</template>
```
:::


### Button combinations

:::demo You can put multiple Buttons in a n-button-group container.
```html
<template>
  <n-button-group>
    <n-button type="primary">Primary 1</n-button>
    <n-button type="primary">Primary 2</n-button>
  </n-button-group>
  <n-button-group>
    <n-button>Default 1</n-button>
    <n-button>Default 2</n-button>
  </n-button-group>
</template>
```
:::


### API

| Property | Description | Type | Default |
| :--- | :--- | :--- | :--- |
| type | can be set to `default`,`primary`,`link`,`ghost`,`delicate` | String | default |
| icon | Icon button, selectable just like the value of the icon | String |  |
| loading    | set the loading status of button | Boolean     | false |
| disabled    | disabled state of button | Boolean     |  |