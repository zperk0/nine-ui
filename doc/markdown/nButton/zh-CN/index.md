# Button 按钮
按钮用于开始一个即时操作。

## 何时使用
标记了一个（或封装一组）操作命令，响应用户点击行为，触发相应的业务逻辑。

## 代码演示

### 按钮类型
:::demo 按钮有五种类型：默认按钮、主按钮、链接按钮、幽灵按钮、微妙按钮。
```html
<template>
  <n-button type="default">
    Default
  </n-button>
  <n-button type="primary" @click="handleClick">
    Primary
  </n-button>
  <n-button type="link">
    Link
  </n-button>
  <n-button type="ghost">
    Ghost
  </n-button>
  <n-button type="delicate">
    Delicate
  </n-button>
</template>

<script>
export default {
  methods: {
    handleClick () {
      alert('Ninecat')
    }
  }
}
</script>

```
:::

### 图标按钮

:::demo 当需要在 Button 内嵌入 Icon 时，可以设置 icon 属性，或者直接在 Button 内使用 Icon 组件。
```html
<template>
  <n-button><n-icon name="icon-home" /></n-button>
  <n-button icon="icon-home" type="primary"/>
</template>
```
:::

### 加载中状态

:::demo 添加 loading 属性即可让按钮处于加载状态，最后两个按钮演示点击后进入加载状态。
```html
<template>
  <n-button :loading="true">loading</n-button>
</template>
```
:::

### 禁用状态

:::demo 添加 disabled 属性即可让按钮处于不可用状态，同时按钮样式也会改变。
```html
<template>
  <n-button :disabled="true">disable</n-button>
</template>
```
:::


### 按钮组合

:::demo 可以将多个 Button 放入 n-button-group 容器中。
```html
<template>
  <n-button-group>
    <n-button type="primary">Primary 1</n-button>
    <n-button type="primary">Primary 2</n-button>
  </n-button-group>
  <n-button-group>
    <n-button>Default 1</n-button>
    <n-button>Default 2</n-button>
  </n-button-group>
</template>
```
:::


## API

| 参数 | 说明 | 类型 | 默认值 |
| :--- | :--- | :--- | :--- |
| type | 按钮类型，可选择值`default`,`primary`,`link`,`ghost`,`delicate` | String | default |
| icon | 图标按钮，可选择和图标的值一样 | String |  |
| loading    | 是否为加载状态 | Boolean     | false |
| disabled    | 禁用状态 | Boolean     | false |