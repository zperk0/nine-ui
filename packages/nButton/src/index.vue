<template>
  <button
    class="nbutton"
    :class="[
      type ? type : '',
      disabled ? 'nbutton-disabled' : ''
    ]"
    :disabled="disabled"
    @click="$emit('click')"
  >
    <span
      v-if="icon || loading"
      class="nbutton-icon"
    >
      <n-icon
        v-if="loading"
        class="loading"
        name="icon-wind-loading"
      />
      <n-icon
        v-if="icon && !loading"
        :name="icon"
      />
    </span>
    <span class="nbutton-content">
      <slot />
    </span>
  </button>
</template>

<script>
export default {
  name: 'NButton',
  props: {
    type: {
      type: String,
      default: '',
      validator (value) {
        return typeof value === 'string'
      }
    },
    icon: {
      type: String,
      default: ''
    },
    loading: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean
    }
  },
  mounted () {
    console.log(this.$props)
  }
}
</script>

<style lang="scss" scoped>

.nbutton{
  padding: 0.5em 1em;
  background-color: #F7F7FA;
  border-radius: 6px;
  border: 0;
  vertical-align: middle;
  cursor: pointer;
  &:focus,&:hover{
    background-color: #E5E5EA;
    outline: none;
  }
  .nbutton-icon{
    display: inline-block;
    margin-right: 0.2em;
  }
}

.nbutton-disabled {
  background-color: #F7F7FA;
  cursor: not-allowed;
  &:focus,&:hover{
    background-color: #F7F7FA;
    outline: none;
  }
}
</style>
