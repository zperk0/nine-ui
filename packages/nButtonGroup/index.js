import NButtonGroup from './src/index.vue'

NButtonGroup.install = function (Vue) {
  Vue.component(NButtonGroup)
}

export default NButtonGroup
