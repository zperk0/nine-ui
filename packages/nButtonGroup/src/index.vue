<template>
  <div class="n-button-group">
    <slot />
  </div>
</template>

<script>
export default {
  name: 'NButtonGroup',
  mounted () {
    for (let node of this.$el.children) {
      const name = node.nodeName.toLowerCase();
      if (name !== 'button') {
        console.warn(`n-button-group 的子元素应该全是 n-button，但是你写的是 ${name}`);
      }
    }
  }
};
</script>

<style lang="scss" scoped>
.n-button-group{
   display: inline-flex;
   vertical-align: middle;
   /deep/ .nbutton{
     border-radius: 0px;
     &:first-child{
       border-top-left-radius: 6px;
       border-bottom-left-radius: 6px;
     }
     &:last-child{
       border-top-right-radius: 6px;
       border-bottom-right-radius: 6px;
     }
   }

}
</style>
