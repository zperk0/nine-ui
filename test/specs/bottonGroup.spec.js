import nButtonGroup from '../../packages/n-button-group';
import nButton from '../../packages/n-button';
import { mount } from '@vue/test-utils';

describe('nButtonGroup', () => {
  it('renders correctly', () => {
    const wrapper = mount({
      render () {
        return <nButtonGroup>TEST</nButtonGroup>;
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('redner slot button correctly', () => {
    const wrapper = mount({
      render () {
        return <nButtonGroup><nButton>Primary</nButton></nButtonGroup>;
      }
    });
    expect(wrapper.text()).toBe('Primary');
  });
  it('redner slot two buttons in this nButtonGroup', () => {
    const wrapper = mount({
      render () {
        return (<nButtonGroup>
          <nButton type="primary">Primary</nButton>
          <nButton type="primary">Primary</nButton>
        </nButtonGroup>);
      }
    });
    expect(wrapper.find('.primary').exists()).toBe(true);
  });
});
