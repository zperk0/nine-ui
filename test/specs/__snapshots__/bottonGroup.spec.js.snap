// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nButtonGroup renders correctly 1`] = `<div class="n-button-group">TEST</div>`;
